### DISCUSSION TITLE
How to determine the weights of locally added fonts

### DISCUSSION THREAD

**Discussion by kakarot-dev** [samizdatco/skia-canvas#65] [Q&A] [answered]
how to set the font weights in the FontLibrary ?

**Comment by samizdatco** [comment, marked as answer]
Weights are determined by the font files themselves so you don't need to specify them when calling [`FontLibrary.use()`](https://github.com/samizdatco/skia-canvas#usefamilyname-fontpaths). Try inspecting the value returned by `use()` to see the numerical weights it derived from the files—it will look something like:
```js
{
  family: 'Grizwald',
  weight: 600,
  style: 'normal',
  width: 'normal',
  file: 'fonts/Oswald-SemiBold.ttf'
}
```

You can also get a list of available weights after you've loaded fonts by calling [`FontLibrary.family()`](https://github.com/samizdatco/skia-canvas#familyname)

**Comment by kakarot-dev** [comment]
Ok so I cant manually set the font weight right?

**Comment by samizdatco** [comment]
Right, the font files themselves define their weights (i.e., you can't *override* those settings). You can use whatever weight you want when you set the [`font`](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/font) value though.

**Comment by quiple** [comment]
@samizdatco When combining multiple font families, I often need to redefine or remap the font weights so they match visually across different fonts. Is there a way to achieve this?

For instance, SF Pro uses non-standard weight values like `[33, 100, 200, 400, 530, 600, 700, 780, 810]`. I need to remap these to the standard scale of `[100, 200, 300, ..., 900]` for better compatibility.

In CSS, This is resolved by specifying the `font-weight` within the `@font-face` rule.
